src/document_store.zig

```diff
@@ -67,8 +62,8 @@ fn newDocument(self: *DocumentStore, uri: []const u8, text: []u8) !*Handle {
         },
     };
     try self.checkSanity(&handle);
-    try self.handles.putNoClobber(uri, handle);
-    return &(self.handles.get(uri) orelse unreachable).value;
+    const kv = try self.handles.getOrPutValue(uri, handle);
```

Comment:
Nice, I somehow missed this function in HashMap...  
I think the errdefer is still needed here though, this function takes ownership of `uri` and `text`.

Reply:
Ah, I guess it is confusing since `duped_text` and `duped_uri` are errdefer free'd in `openDocument`.  
This is a bug, they should be handled by `newDocument`.

Reply:
it's very odd to take ownership of arguments if an error is thrown.

Reply:
Yes, I see your point.  
I think this way is clearer for the caller so it should be preferred.  
Thanks for your excellent PR and your input :)